bsnes/ui-qt/debugger/tools/qhexedit2/qhexedit.cpp

```diff
@@ -965,3 +996,57 @@ void QHexEdit::updateCursor()
     viewport()->update(_cursorRect);
 }
 
+void QHexEdit::updateAnimatedHexValues()
+{
+    if (_trackMemoryChanges && !_hexChangesMap.keys().empty()) {
+        QHash<qint64, quint16>::iterator it = _hexChangesMap.begin();
+        while (it != _hexChangesMap.end()) {
+           it.value()--;
+           if (it.value() == 0) {
+               it = _hexChangesMap.erase(it);
+           } else {
+               ++it;
+           }
+        }
+
+        viewport()->update();
+    }
+}
+
+QHexEdit::ScopedMemoryTracker::ScopedMemoryTracker(QHexEdit *editor)
+    : _editor(editor){
+    Q_ASSERT(_editor);
+    const QByteArray currentData = _editor->_dataShown;
+    const qint64 first = _editor->_bPosFirst;
+    const qint64 last = _editor->_bPosLast;
+    const qint64 editorSize = _editor->_editorSize;
+
+    for (qint64 i = first; i < qMin(editorSize, last + BYTES_PER_LINE + 1); i++) {
+        const qint64 addr = i + _editor->_addressOffset;
+        quint16 value = qFromLittleEndian<quint16>(currentData.mid(i-first, 1).data());
```

Comment:
Why's this making int16s from single bytes? This should probably just be something like `char value = currentData[i-first]`, with _dataHash's type changed to match.

Reply:
Nice catch, I think I must have accidentally left that in there from a premature optimization. I've fixed it.